bitcoin/src/p2p/mod.rs

```diff
@@ -24,4 +25,6 @@ pub mod message_filter;
 #[cfg(feature = "std")]
 pub mod message_network;
 
-pub use self::constants::Magic;
+pub use self::constants::{ServiceFlags, PROTOCOL_VERSION};
+/// Re-export so users can `use bitcoin::p2p::Network`.
+pub use self::network::Network;
```

Comment:
Oh, so it is re-exported but not used in our code...

Reply:
After #1880 I think we should go through and mechanically investigate every symbol that appears to be re-exported from multiple places, and try to give it a single canonical home (plus the crate root, perhaps).

If it were possible to deprecate re-exports (I don't think we can?) that would make this much easier.

Reply:
So the re-export here was intentional, the reasoning is as follows. For usecase of `Network` type on its own, this is best
```rust
use bitcoin::Network
```
For usecase of multiple types in network module this is best:
```rust
use bitcoin::p2p::network::{ChainHash, Network}
```
And for usecase where devs want more context inline this is best:
```rust
use bitcoin::p2p;
use some::other::foo;
...
let n = p2p::Network;
let m = foo::Bar;
```

Or am I overthinking it?

Reply:
Note this is not how we do things anywhere else in the code, I started a discussion and deleted it. Leaving this discussion for closer to v1.0 release.

Reply:
Yeah, sounds right. `network::Network` is not re-export but feels like the right place for `Network` to live in.